### PR TITLE
feat(apikey): added route to list user api keys

### PR DESCRIPTION
**Pull request by peusebiu**
adding api key expiration date

**What type of PR is this?**
feature

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:
UI needs a way to get list of API keys for the current user, added route for this.
Added option to expire API keys after a certain date.

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -327,31 +327,113 @@ Create an API key for the current user using the REST API
 **Sample input**:
 
 ```
-POST /auth/apikey
-Body: {"label": "git", "scopes": ["repo1", "repo2"]}'
+POST /auth/apiKey
```
Comment (peusebiu) [inline examples/README.md:330]: todo: fix typo

File: pkg/api/constants/consts.go [hunk 2]
```diff
@@ -20,4 +22,5 @@ const (
 	SessionClientHeaderValue     = "zot-ui"
 	APIKeysPrefix                = "zak_"
 	CallbackUIQueryParam         = "callback_ui"
+	APIKeyTimeFormat             = time.RFC1123Z
```
Comment (peusebiu) [inline pkg/api/constants/consts.go:25]: let's decide what format we use for expiration date.

This format will be used by API clients to set expiration date on an API key.

**Review by andaaron — APPROVED** [review]
LGTM

File: pkg/api/routes.go [hunk 3]
```diff
@@ -2029,8 +2030,50 @@ func (rh *RouteHandler) GetOrasReferrers(response http.ResponseWriter, request *
...
+// GetAPIKeys godoc
+// @Summary Get list of API keys for the current user
+// @Description Get list of all API keys for a logged in user
+// @Accept  json
+// @Produce json
+// @Success 200 {string} string "ok"
+// @Failure 401 {string} string "unauthorized"
+// @Failure 404 {string} string "not found"
+// @Failure 500 {string} string "internal server error"
+// @Router /auth/apikey  [post].
```
Comment (andaaron) [inline pkg/api/routes.go:2047]:
```suggestion
// @Router /auth/apikey  [get].
```

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**